syntax = "proto3";

import "google/protobuf/empty.proto";

package product;

option java_multiple_files = true;
option java_package = "com.example.product";
option java_outer_classname = "ProductProto";

service ProductService {
  rpc ListProducts (google.protobuf.Empty) returns (ListProductsResponse);
  rpc GetProduct (GetProductRequest) returns (Product);
  rpc CreateProduct (CreateProductRequest) returns (Product);
}

message Product {
  string id = 1;
  string name = 2;
  string description = 3;
  double price = 4;
}

message ListProductsResponse {
  repeated Product products = 1;
}

message GetProductRequest {
  string id = 1;
}

message CreateProductRequest {
  string name = 1;
  string description = 2;
  double price = 3;
}
